### ISSUE TITLE
V5.0 testing C++ Perspective cannot be extended - esp. with build Configuration

### ISSUE THREAD

**Issue by ArminJo** [Sloeber/arduino-eclipse-plugin#1693]
Is it intended to not support build configurations?
I enabled it here
![grafik](https://github.com/user-attachments/assets/8fbc6ff4-70ba-4ea2-896f-6144a5ffb953)
but it gets unchecked after I apply and close the perspective setting window and open it again :-(

  Sloeber, the Eclipse Arduino IDE (plugin version)	5.0.0.202411250122

**Comment by jantje** [comment]
This is a part of Eclipse CDT I didn't even know it existed.
So nothing is intended from my part.

**Comment by ArminJo** [comment]
So I have hope, that this will be fixed in the future 😀.
For now I can test without it, but as you may remeber I use the configuration heavily for "production".
![grafik](https://github.com/user-attachments/assets/250be30c-d503-4e5d-809d-4982edb3f5b2)

**Comment by jantje** [comment]
V5 is supposed to be way better on behalf of configurations (especially config changes)
But I have no clue what is missing ;-)
I'll take a look.

**Comment by jantje** [comment]
I've taken a look and I do not understand why configuration is not visible.
As a workaround you can select the c/c++ perspective to start from.
There is no real difference.

**Comment by jantje** [comment]
Strange ... I went to the C/C++ perspective activated the "build configurations" played around went back to sloeber perspective and there is was.
![image](https://github.com/user-attachments/assets/96fed5b3-4cc0-48ef-b318-47760e188c4b)

**Comment by jantje** [comment]
This clearly seems to be an eclipse issue that I can not be fix nor do I know who could.

**Comment by ArminJo** [comment]
With CDT (also latest updates) I have no such problems...

**Comment by jantje** [comment]
I understand but Build configuration is CDT stuff.
It looks as if part of CDT is not loaded and after looking at in in CDT perspective it gets loaded and is available.
(Eclipse uses lazy loading which mean stuff is only loaded when needed; it might very well be that this CDT gui part is not loaded when Sloeber starts and does not reference these perspective contributors)

**Comment by ArminJo** [comment]
Ok, one year is gone and I made further tesing because I want to set up a remote Sloeber system.

If I install Eclipse CDT using eclipse-cpp-2026-03-R-win32-x86_64.zip and add "install new Software" >Repository.zip, I can open a workspace and see this:
<img width="896" height="99" alt="Image" src="https://github.com/user-attachments/assets/395fdc38-6be5-42d0-bd40-edbcf2e92bf8" />



If I install Sloeber directly using latest sloeber-ide-sloeber_release-win32.win32.x86_64.zip I see this on opening the same workspace:
<img width="725" height="97" alt="Image" src="https://github.com/user-attachments/assets/bcc5e6f7-0511-40b7-a539-ad7c109f47d9" />



I can switch between the two Sloeber versions and the buid configurations come and go, so it only depends on the Slober "binary" running, not the workspace.
As far as I remember the same holds for an empty or small workspace.

So what is the subtle difference in perspective handling between these two installations?
AFAIK the perspective data is stored  is the file WORKSPACE_....\.metadata\.plugins\org.eclipse.e4.workbench\workbench.xmi

**Comment by jantje** [comment]
To understand this you need to understand gui elements and eclipse products.

### Gui elements:
Eclipse gui is made up of gui elements.
Gui elements are provided by plugins and identify themselves to eclipse in the plugin.xml using id's (like org.eclipse.ui.menus) pointing to classes, images and **perspectives**.
The idea is that the "user" selects the gui elements to show.
However requiring every user to select all gui elements is unrealistic so users select a perspective (which is actually only a ID and some house keeping info) but the user can modify these perspectives fully.
As stated the perspective is only an ID (like io.sloeber.product.perspective ) it's content is actually provided by gui elements who "identify as perspective components"

<img width="1185" height="511" alt="Image" src="https://github.com/user-attachments/assets/a52cc742-4e32-47ec-a83e-7dccc1f022a5" />

As you can see plugins can provide perspectives themselves (sloeber, debug ...) to make it easier for the user.
Because of this flexibility; when gui elements are not (fully -yes it is a mess-) available they will not be shown.
So simply adding or removing a plugin will change the gui.
Changing the perspective changes the gui (potentially completely like going from sloeber to debug)

So to have 100% the same gui in 2 different installs they have to have the exact same plugins installed (maybe also in the same version) and be in the same perspective and the perspective needs to be the default one.
If you start from different eclipse products this is very unlikely the case.

### eclipse products
Once you have created a plugin you can distribute your plugin via "install new software".
However if your plugin is dependent on "other stuff" (like CDT) and comes best with "more stuff" (like egit) there is no way to guarantee these functionalities (read plugins and as such gui elements) will be available leading to "unsatisfied customers".
Because of this Sloeber comes as a product (and we officially only support the product).
In other words: A product allows you to bundle your plugin with other plugins and wrap a nice tie around it (like the splash screen and the toolbar icon)

<img width="662" height="633" alt="Image" src="https://github.com/user-attachments/assets/13528f33-7160-430f-8665-49b592bb2047" />

### What does that mean for you
In your case you installed 2 different products (CDT product versus Sloeber product) so you will have different plugins and as such different gui elements.
You are likely to have a different perspective as well (sloeber versus CDT).
What I'm trying to say is that it is unrealistic to expect the same gui view "out of the box" when starting from 2 different products but you should be able to get to it.
I hope this helps.

**Comment by ArminJo** [comment]
Thank you for your elaborated explanation 👍 

As far as I understand now, the CDT provides the buid configurations plugins, while Sloeber is missing them.
So is it possible for you to also add this plugin(s) to Sloeber 5, as in Sloeber 4.4?
Otherwise build configuration is not supported any more in "native" Sloeber 5 :-( not even Debug and Release.
You can modify build configuration settings in Sloeber but you cannot select one as active :-(.

As I can see Sloeber 5 is especially missing these features:
Eclipse CDT	CDT CMake UI	12.4.0.202512101642	org.eclipse.cdt.cmake
and
Eclipse CDT	LaunchBar UI	12.4.0.202512101642	org.eclipse.launchbar

Maybe there is a chance, that adding one or both of these, will bring back the build configuration functionality 😀 .

**Comment by jantje** [comment]
I'm not sure why you think it is missing as I have been using this on a daily basis.

<img width="801" height="664" alt="Image" src="https://github.com/user-attachments/assets/01df8ba3-546a-4544-ae43-c674fe11520e" />
So I guess something else is missing (the configuration stuff is part of the CDT core plugin and do not start about it unless you want to get a ton load of shit about how shit the whole configuration thing is implemented in CDT)

>Eclipse CDT CDT CMake UI 12.4.0.202512101642 org.eclipse.cdt.cmake

I can add to the sloeber product if I can find it 😕

<img width="486" height="429" alt="Image" src="https://github.com/user-attachments/assets/75ad995c-0fce-431c-bf3f-36daf753debe" />

>Eclipse CDT LaunchBar UI 12.4.0.202512101642 org.eclipse.launchbar

I doubt the CDT launchbar is called org.eclipse.launchbar and as you can see above there is no org.eclipse.cdt.launchbar in my current Sloeber development environment (which basically means: there is no link to a repository that contains this info in the sloeber target file in the sloeber project)

**Comment by ArminJo** [comment]
Thanks!
I swear, I never in my whole life used the menu entry to access these functions, so I forgot about this possibility, Sorry!

**Comment by jantje** [comment]
> I never in my whole life used the menu entry to access these functions

So how did you do it then?

**Comment by ArminJo** [comment]
I always used the GUI Icons, and therefore I was so desperate not having them in Sloeber ...

**Comment by jantje** [comment]
I think the launchbar is already "legacy" but I may be wrong.
Another way (though longer but maybe handy when you are there already) is in project properties.
In all CDT related pages that configure configuration setting you have a dropdown box containing the configs and a configure.
In the configure you can set the active config.
